core/sail/base/src/main/java/org/eclipse/rdf4j/sail/base/SailSourceConnection.java

```diff
@@ -135,10 +141,17 @@
 	 * @param resolver
 	 */
 	protected SailSourceConnection(AbstractSail sail, SailStore store, FederatedServiceResolver resolver) {
+		this(sail, store, new SimpleEvaluationStrategyFactory(resolver), resolver);
```

Comment:
Historical artifact. Ideally, there would be no sight of FederatedServiceResolver here as the EvaluationStrategyFactory is responsible for manuafacturing fully configued EvaluationStrategies. But you are right, I can do a better job here.